src/middleware.js

```diff
@@ -1,8 +1,13 @@
-import { BACK, FORWARD, GO, PUSH, REPLACE } from './action-types.json';
+import { BACK, FORWARD, GO, PUSH, REPLACE, UPDATE } from './action-types.json';
+
+const isEqual = (object1, object2) => JSON.stringify(object1) === JSON.stringify(object2);
 
 export default (router) => () => (next) => (action) => {
 	switch (action.type) {
 		case PUSH: return router.push(...action.payload);
+		case UPDATE:
+			const query = { ...router.currentRoute.query, ...action.payload[0] };
+			return !isEqual(query, router.currentRoute.query) ? router.push({ query }) : next(action);
```

Comment:
Do we care if the routes are equal or not? Could this be simplified to:
```
case UPDATE: return router.push({ query: action.payload[0] }, ...action.payload.slice(1));
```

Reply:
The purpose was to maintain the unchanged query values on the route- so at minimum we should use {query: { ...router.currentRoute.query, ...action.payload[0]}} - isEqual test was to avoid duplicatedPath navigation errors- which also occur when ...action.payload.slice(1) is also provide on the push . If we do not intend to keep track of the current query strings parameters - we can just use the existing push action -- I will see if I can accomplish the same thing in a thunk - and avoid making changes to this package

Reply:
That makes sense. Should be fine:
```
case UPDATE: return router.push({ query: { ...router.currentRoute.query, ...action.payload[0] }, ...action.payload.slice(1) } });
```